framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -4631,6 +4613,442 @@ private void makeGroundTargetType(
+   * for all the details.
+   *
+   * <p>If {@code type} is not a declared type or if it does not have any wildcard type arguments,
+   * this method returns {@code type}.
+   *
+   * @param typeToCapture type to capture
+   * @return the captured type
+   */
+  public AnnotatedTypeMirror applyCaptureConversion(AnnotatedTypeMirror typeToCapture) {
+    TypeMirror capturedTypeMirror = types.capture(typeToCapture.getUnderlyingType());
+    return applyCaptureConversion(typeToCapture, capturedTypeMirror);
+  }
+
+  /**
+   * Apply capture conversion.
+   *
+   * <p>Capture conversion is the process of converting wildcards in a generic type to fresh type
+   * variables. See <a
+   * href="https://docs.oracle.com/javase/specs/jls/se11/html/jls-5.html#jls-5.1.10">JLS 5.1.10</a>
+   * for all the details.
+   *
+   * <p>If {@code type} is not a declared type or if it does not have any wildcard type arguments,
+   * this method returns {@code type}.
+   *
+   * @param type type to capture
+   * @param typeMirror type from the Java compiler that has captured wildcards, which are used as
+   *     the underlying type of the captured wildcards
+   * @return the captured type
+   */
+  public AnnotatedTypeMirror applyCaptureConversion(
+      AnnotatedTypeMirror type, TypeMirror typeMirror) {
+    if (!shouldCapture(type, typeMirror)) {
+      if (type.containsUninferredTypeArguments()
+          && typeMirror.getKind() == TypeKind.DECLARED
+          && type.getKind() == TypeKind.DECLARED) {
+        // If capture conversion is not applied because the type contains uninferred
+        // wildcards, then mark any uncaptured wildcards as "uninferred".
+        DeclaredType capturedTypeMirror = (DeclaredType) typeMirror;
+        AnnotatedDeclaredType uncapturedType = (AnnotatedDeclaredType) type;
+        for (int i = 0; i < capturedTypeMirror.getTypeArguments().size(); i++) {
+          TypeMirror capturedTypeArgTM = capturedTypeMirror.getTypeArguments().get(i);
+          AnnotatedTypeMirror uncapturedTypeArg = uncapturedType.getTypeArguments().get(i);
+          if (uncapturedTypeArg.getKind() != TypeKind.WILDCARD) {
+            continue;
+          }
+          if (TypesUtils.isCaptured(capturedTypeArgTM)
+              || capturedTypeArgTM.getKind() != TypeKind.WILDCARD) {
+            ((AnnotatedWildcardType) uncapturedTypeArg).setUninferredTypeArgument();
+          }
+        }
+      }
+      return type;
+    }
+
+    DeclaredType capturedTypeMirror = (DeclaredType) typeMirror;
+    AnnotatedDeclaredType uncapturedType = (AnnotatedDeclaredType) type;
+    final AnnotatedDeclaredType capturedType =
+        (AnnotatedDeclaredType) AnnotatedTypeMirror.createType(capturedTypeMirror, this, false);
+
+    captureTypeArgCopier.copy(uncapturedType, capturedType);
+
+    AnnotatedDeclaredType typeDeclaration =
+        (AnnotatedDeclaredType) getAnnotatedType(uncapturedType.getUnderlyingType().asElement());
```

Comment:
What is the relationship between `uncapturedType.getUnderlyingType()` on this line and local variable `capturedTypeMirror`?